pkg/solana/chainreader/chain_reader.go

```diff
@@ -187,7 +187,9 @@ func (s *ContractReaderService) GetLatestValue(ctx context.Context, readIdentifi
 
 	// TODO this is a temporary edge case - NONEVM-1320
 	if values.reads[0].readName == GetTokenPrices {
-		return s.handleGetTokenPricesGetLatestValue(ctx, params, values, returnVal)
+		if err := s.handleGetTokenPricesGetLatestValue(ctx, params, values, returnVal); err != nil {
```

Comment:
Wouldn't this returnVal get overwritten by the batch call? I could be missing something though since the context wouldn't be in the PR.

Reply:
I probably shouldn't have referred to it as a batch call but I meant just the `doMethodBatchCall` call below. But think I just answered my own question looking at this again.